addons/survey/models/survey_survey.py

```diff
@@ -581,11 +581,30 @@ def _get_pages_or_questions(self, user_input):
             if self.questions_selection == 'random' and not self.session_state:
                 result = user_input.predefined_question_ids
             else:
-                result = self.question_and_page_ids.filtered(
-                    lambda question: not question.is_page or not is_html_empty(question.description))
+                result = self._get_pages_and_questions_to_show()
 
         return result
 
+    @api.model
```

Comment:
Not sure why api.model as this clearly uses self?

Reply:
Indeed, this was leftover from a previous version